process/block/baseProcess.go

```diff
@@ -724,13 +727,22 @@ func (bp *baseProcessor) cleanupBlockTrackerPools(headerHandler data.HeaderHandl
 }
 
 func (bp *baseProcessor) cleanupBlockTrackerPoolsForShard(shardID uint32, noncesToFinal uint64) {
-	selfNotarizedNonce := bp.forkDetector.GetHighestFinalBlockNonce()
+	shardForSelfNotarized := bp.getShardForSelfNotarized(shardID)
+	selfNotarizedHeader, _, err := bp.blockTracker.GetLastSelfNotarizedHeader(shardForSelfNotarized)
+	if err != nil {
+		log.Trace("cleanupBlockTrackerPoolsForShard.GetLastSelfNotarizedHeader",
```

Comment:
why not log.Warn ? This might be an error if I cannot see my lastselfnotarizedheader

Reply:
Done